pkg/monitoring/domainstats/downwardmetrics/scraper.go

```diff
@@ -146,9 +147,12 @@ type Collector struct {
 }
 
 func NewReporter(nodeName string) *DownwardMetricsReporter {
+	// Let's return a hash of the host name to avoid exposing the real host name to an
+	// untrusted VM. The metrics `HostName` value is used as an uuid of the node (to
+	// detect changes), and is not used to do dns queries.
 	return &DownwardMetricsReporter{
 		staticHostInfo: &StaticHostMetrics{
-			HostName:             nodeName,
+			HostName:             util.HashString(nodeName),
```

Comment:
I do not think that we can do this, as it is breaking the API (same key, in future different value).
And there are use-cases to expose this value.

Did you consider to expose the host ip? IIRC then this is "public information" as it's provided on the POD status.

Reply:
I get your point @fabiand. The way to do thing right would be to have a configuration object describing how the outputted information should look like, e.g.

`exposeCleartextHostname = false (per default [1])`

The `staticHostMetrics` object should be

```
if exposeCleartextHostname == false:
    HostName:  util.HashString(nodeName)
else:
    HostName:  nodeName
```
WDYT? @fabiand @germag ?


[1] which would still break backwards compatibility but AFAIK at least for OCPv there is no production workload using this. And for folks wanting to keep compatibility to the vhostmd format could flip `exposeCleartextHostname` to `true` .

Reply:
The VM user might be not allowed to get any information about the cluster, or the VM user might want to get the relation between the downward metrics to other metrics from the cluster. Both views seems to be valid. For this reason, I would vote to implement the configuration like @newkit proposed, but this might delay the PR until https://issues.redhat.com/browse/CNV-38564 is addressed.

Reply:
> If we make the value of hostname cnfigurable, where would this config be?

The config should be on cluster level, next to the config to enable the downward metrics feature itself.
This way the cluster admin can control, which information is exposed to the VM users.

Reply:
@fabiand Would you agree with the approach to introduce a config value to enable the feature, and if the feature is enabled, the cluster admin could configure on cluster level if hashing is enabled, if this hashing is refgelcted in the metrics itself like this:
```
  <metric type='string' context='host'>
    <name>HostName</name>
    <value hashed='false'>vm0</value>
  </metric>
```
or
```
  <metric type='string' context='host' >
    <name>HostName</name>
    <value hashed='true'>965c80ad937198db8ebb6edea7be1ece6dd15d997665179126f5844e7f3e9671</value>
  </metric>
```

Reply:
To me this would do the trick, yes!

Reply:
Are the SAP eng. ok with adding an attribute to that element?, I don't think anyone besides SAP uses this xml but just in case someone is parsing this manually(*)

(*) I know is not a problem for a proper xml parser, but you never know :) [Parsing XML The Cthulhu Way ](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454)